lib/private/DB/QueryBuilder/FunctionBuilder/FunctionBuilder.php

```diff
@@ -45,8 +45,13 @@ public function md5($input): IQueryFunction {
 		return new QueryFunction('MD5(' . $this->helper->quoteColumnName($input) . ')');
 	}
 
-	public function concat($x, $y): IQueryFunction {
-		return new QueryFunction('CONCAT(' . $this->helper->quoteColumnName($x) . ', ' . $this->helper->quoteColumnName($y) . ')');
+	public function concat($x, ...$expr): IQueryFunction {
+		$args = func_get_args();
+		$list = [];
+		foreach ($args as $item) {
```

Comment:
```suggestion
	public function concat(...$expr): IQueryFunction {
		$list = [];
		foreach ($expr as $item) {
```

Is the same and looks better?

Reply:
https://github.com/nextcloud/server/pull/30378#discussion_r775555784